cfme/services/service_catalogs/ui.py

```diff
@@ -73,9 +73,12 @@ class DetailsServiceCatalogView(ServicesCatalogsView):
 
     @property
     def is_displayed(self):
+
         return (
-            self.in_service_catalogs and self.service_catalogs.is_opened and
-            self.title.text == 'Service "{}"'.format(self.context['object'].name)
+            self.in_service_catalogs and
+            wait_for(lambda: self.title.text == 'Service "{}"'.format(self.context['object'].name),
```

Comment:
**Required** Having a `wait_for` in a `is_displayed` property is not a good practice, and shouldn't be necessary with methods like `wait_displayed` that can be called from the view object.

Reply:
@john-dupuy : The title of page is sometimes taking wrong value . I printed it and saw "All Service Catalog Items" when it is in service page . It changes soon but test fails at the first time itself . UI shows correct title . I have used wait_dispalyed in view but that's not fixing it either .

===========self.in_service_catalogs=========
True
All Service Catalog Items
=====self.title.text==============
Service "bz_1705021"
---------submit--------
False
===========self.in_service_catalogs=========
True
Service "bz_1705021"
=====self.title.text==============
Service "bz_1705021"
---------submit--------